src/App.jsx

```diff
@@ -3,26 +3,92 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { request } from './api/api';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    titleFilter: '',
+    todosStatus: 'All',
   };
 
+  componentDidMount() {
+    request('/todos')
+      .then(({ data }) => this.setState({ todos: [...data] }));
+  }
+
+  titleFilterChangeHandler = ({ name, value }) => {
+    this.setState({ [name]: value });
+  }
+
+  setTodosStatus = (status) => {
+    this.setState({ todosStatus: status });
+  }
+
+  clearUser = () => {
+    this.setState({ selectedUserId: 0 });
+  }
+
+  selectUser = (userId) => {
+    this.setState((state) => {
+      if (state.selectedUserId === userId) {
+        return state;
+      }
+
+      return {
+        ...state,
+        selectedUserId: userId,
+      };
+    });
+  }
```

Comment:
More reliable way is to make `setState` call conditional:
```suggestion
  selectUser = (userId) => {
    if (state.selectedUserId !== userId) {
      this.setState({ selectedUserId: userId });
    }
  }
```